solutions/cses-1640.mdx

```diff
@@ -41,132 +40,173 @@ One idea that comes to mind would be to used a boolean array to store the
 values. Unfortunately, since $a_i \leq 10^9$, this approach isn't feasible.
 
 <LanguageSection>
-
-<PySection>
-
-However, we can store the values in a dictionary which maps each value to an
-index, and use the
-[`__contains__`](https://stackoverflow.com/questions/1964934/what-does-contains-do-what-can-call-contains-function)
-dunder method to check whether a value exists, and return the corresponding
-index if it does.
-
-</PySection>
-
 <CPPSection>
 
 However, we can store the values in an (un)ordered map which maps each value to
 an index, and use the `.count` method to check whether a value exists, and
 return the corresponding index if it does.
 
 </CPPSection>
-
 <JavaSection>
 
 However, we can store the values in a map which maps each value to an index, and
 use the `.containsKey` method to check whether a value exists, and return the
 corresponding index if it does.
 
 </JavaSection>
+<PySection>
 
+However, we can store the values in a dictionary which maps each value to an
+index. Then, we can use the dictionary to check in $\mathcal{O}(1)$ time get the corresponding
+index for a value (if there is one).
+
+</PySection>
 </LanguageSection>
 
 To be careful not to count the same index twice, we'll add values to the map as
 we iterate through it, so at some index $i$ you only consider values with index
 $j < i$.
 
 <LanguageSection>
-
 <CPPSection>
 
 ```cpp
-//CodeSnip{CPP Short Template}
+#include <bits/stdc++.h>
+using namespace std;
 
 int main() {
-	int n, x; cin >> n >> x;
-	vi a(n);
-	map<int, int> vals;
+	int n, x;
+	cin >> n >> x;
+
+	vector<int> values(n);
 	for (int i = 0; i < n; i++) {
-		cin >> a[i];
+		cin >> values[i];
 	}
+	
+	// use a map to avoid using a very large array
+	map<int, int> val_to_ind;
```

Comment:
any specific reason to use `map` instead of `unordered_map`? 
u don't necessarily have to change it, but in Java we also used a hashmap

Reply:
in c++ i just use map as default
it's actually faster than unordered_map sometimes